delivery_service.rb

```diff
@@ -0,0 +1,40 @@
+class DeliveryService
+  attr_accessor :cars_count, :bikes_count, :auto_park
+
+  require_relative 'car'
+  require_relative 'bike'
+  require_relative 'constants'
+
+  def initialize(cars_count = 6, bikes_count = 8)
+    @auto_park = []
+    cars_count.times do
+      @auto_park << Car.new
+    end
+    bikes_count.times do
+      @auto_park << Bike.new
+    end
+    @auto_park.sort_by! { |a| a.speed }
+  end
+
+  def find_transport(distance)
+    free = false
+    free_trans = @auto_park.find { |a| a.available == true && a.distance >= distance }
```

Comment:
What about max weight in this finding method?

Reply:
What about giving priority to `Bike` instance here?

Reply:
'weight' added, mathod separated.

Reply:
Array **autopark** sorted by speed descending, so available bike will be get first. Sorting is easier and it works. Or no?

Reply:
I understand, thanks for explanation